### PR TITLE
Bg testing

### PR DESCRIPTION
**Pull request by fabriceumuhire**
### What is this PR about
- Creation and perform of API Tests
### How does it work
- Create tests and run using mocha and nyc for coverage
- Run local tests on Travis CI and observe coverage on coveralls
- Create a README badge based on coverall markdown
### How to check
- From branch, display of coverage percentage is shown

### CODE REVIEW

File: server/src/test/articles.test.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      chai.request(server)
+        .get(`/api/routes/articles/${artID}`)
+        .end((error,res) => {
+          res.should.have.status(200);
+          //res.body.should.be.a("array");
+        done();
+        });
+    });
+    /* it("It should not GET article", (done) => {
+      const artID = "5f65ad0bca66312e5c555ee";
```
Comment (niyodusengaclement) [inline server/src/test/articles.test.js]: Delete commented linea

File: server/.nyc_output/01843cbe-8b86-4b05-9ac9-8b6ccaa3a6ac.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (niyodusengaclement) [inline server/.nyc_output/01843cbe-8b86-4b05-9ac9-8b6ccaa3a6ac.json:1]: U should not push nyc output

File: server/src/test/articles.test.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      }
+      chai.request(server)
+        .post("/api/routes/articles")
+        .field(article)
+        .attach("image", `${path.join(__dirname,'../uploads/img/ideas.jpg')}`)
+        .type("form")
+        .end((error,res) => {
+          res.should.have.status(401);
+          //res.body.should.have.property("message").equall("Not authorization");
+          //res.body.should.be.a("array");
```
Comment (niyodusengaclement) [inline server/src/test/articles.test.js]: No commented lines

File: server/src/test/articles.test.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          res.should.have.status(401);
+          //res.body.should.be.a("array");
+        done();
+        });
+    });
+    /* it("It should UPDATE a new article with auth", (done) => {
+      const artID = "5f70b52b6d7e9b1a78af7250";
+      const newArticle = ({ "title": "JS Performance solution- Updaters", "content": "Updaters: Postman is a scalable API testing tool that quickly integrates into CI/CD pipeline." })
+      chai.request(server)
+        .patch(`/api/routes/articles/${artID}`)
```
Comment (niyodusengaclement) [inline server/src/test/articles.test.js]: No commented lines

**Review by niyodusengaclement — COMMENTED** [review]
Delete all commented lines

File: server/.nyc_output/01843cbe-8b86-4b05-9ac9-8b6ccaa3a6ac.json (hunk 2 above)
Comment (fabriceumuhire) [reply server/.nyc_output/01843cbe-8b86-4b05-9ac9-8b6ccaa3a6ac.json:1]: Done

**Pull request merged**